docs/android/features/base-component/customize-views.md

```diff
@@ -0,0 +1,348 @@
+import TimeIcon from '@site/src/components/TimeIcon';
+import Prerequisites from '@site/docs/android/features/_shared/Prerequisites.md';
+import LinkToCustomization from '@site/docs/android/features/_shared/LinkToCustomization.md';
+import * as BaseComponentView from './_component';
+
+# Customize base component views
+
+Basic components are available through the SDK. 
+The aim is to save time by override a template once and reusing it without further configuration.
+
+<Explanation />
+
+## Ingredients
+
+There is 7 differents components that you can override once and will be reuse through the SDK  
+
+## Small components
+
+### Counter
+<BaseComponentView.Counter platform="android"/>
+
+In Order to create your own counter you have to create class that implement `Counter`
+
+```kotlin
+import ai.mealz.sdk.components.baseComponent.counter.Counter
+import ai.mealz.sdk.components.baseComponent.counter.CounterParameters
+
+class MyCustomCounter: Counter {
+
+     @Composable
+    override fun Content(params: CounterParameters) {
+        // Your custom design here 
+    }
+}
+```
+
+Then you can use all arguments we give you thinks to `CounterParameters`
+
+```kotlin
+class CounterParameters(
+    val initialCount: Int?, // initial counter state
+    val isDisable: Boolean = false, // Will be true when counter button need to be disable
+    val onCounterChanged: (newValue: Int) -> Unit, // to call when counter value change
+    val lightMode: Boolean = false, // deprecated
+    val minValue: Int? = null, // value under which you should not go
+    val maxValue: Int? = null, // value above witch you should not go
+    val isLoading: Boolean = false, // should show a loding staus of counter
+    val key: Any? = null // provide an unique key to avoid compose recomposition mistake 
+)
+```
+
+Final step is to set Template in your template configuration: 
+
+```kotlin
+class MealzTemplateManager {
+    init {
+        MiamTheme.Template {
+            defaultViews {
+                counter {
+                    view = MyCustomCounter()
+                }
+            }
+        }
+    }
+}
+```
+
+### LikeButton
+<BaseComponentView.Like platform="android"/>
+
+Like button have two sub templates loading and success
+
+#### LikeButtonLoading
+
+When a user click on like button a api call is launch then we need to block user interaction and we 
+put like button in loading state.
+
+To override this loading state you have to create class that implement `LikeButtonLoading`
+
+```kotlin
+import ai.mealz.sdk.components.baseComponent.likeButton.loading.LikeButtonLoading
+
+
+class MyLikeButtonLoading: LikeButtonLoading {
+
+     @Composable
+    override fun Content() {
+        // Your custom design here 
+    }
+}
+```
+
+:::warning
+There is no parameter in like button loading Content fonction
+:::
+
+:::note
+Fetching or updating like can be quick in order to avoid flikering we recommande 
+making this view looking like not like view
```

Comment:
```suggestion
Fetching or updating the like status is quick so in order to avoid flickering we recommand
making this view the same or similar to the isLiked View.
```